articles/openshift/oshift-how-add-domains-proxy-whitelist.md

```diff
@@ -0,0 +1,130 @@
+services: openshift
+author: Ben Bacon
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Permit outbound internet access in deployments with government community network connectivity
+toc_fullpath: How To/oshift-how-add-domains-proxy-whitelist.md
+toc_mdlink: oshift-how-add-domains-proxy-whitelist.md
+---
+
+# How to permit outbound access to internet hosts in deployments with government community network connectivity
+
+## Overview
+
+This article outlines the necessary steps to whitelist domain names within the Squid proxy. Squid is an enabled service within v3.11 OpenShift clusters deployed with connectivity to government community networks (such as: HSCN, PSN, Janet). Whitelisted domains can be accessed on the internet via the proxy, enabling you to request external resources/data on nodes that previously only had connectivity to government community networks.
+
+## Prerequisites
+
+To complete the steps in this guide, you must have access to a cluster that is running OpenShift Container Platform v3.11 (or newer) with connectivity to a government community network. Squid proxy is not currently enabled within environments that do not have access to a government community network.
+
+You must also have access to either:
+
+- the OpenShift Web console
+
+- `oc`, the OpenShift command-line client (CLI). For more information, see OpenShift's [*Get Started with the CLI*](https://docs.openshift.com/container-platform/3.11/cli_reference/get_started_cli.html)
+
+## Destination domain whitelist
+
+Squid proxy is installed as a service on the control plane load balancers. Within the Squid configuration, we have specified an ACL (Access Control List) as being a list of destination domains (present in a file on the file system), which we will refer to as the whitelist for the remainder of this article. Outbound traffic from the cluster's internal network will pass through the proxy with the destination domain being compared against the whitelist for each connection attempt; if the domain exists then outbound internet traffic is permitted, otherwise it is denied with a HTTP 403 error.
+
+A scheduled job (which runs at 0 minutes past every hour) on the OpenShift cluster Bastion host reads a Config Map named `proxy-whitelist` within the `whitelist` project. If there are any modifications to this Config Map, the job overwrites the previous custom entries within the whitelist and triggers a reconfigure task on the Squid proxy to enable the updated domains to be accessed.
+
+### Assigning non cluster-admin users rights to edit the whitelist 
+
+By default, only users who have been assigned the cluster-admin role will be able to view/edit the `proxy-whitelist` ConfigMap within the `whitelist` project. To allow non cluster-admin users to edit this object, use the following Role Based Access Control (RBAC) commands:
```

Comment:
I think we should have a comment here indicating the power that this is giving to individual users - e.g. to be able to enable access cluster wide to new resources. Effectively an administrative function, we want people to think before they grant this access, it shouldn't be handed out to all cluster users.

Reply:
This wasn't quite what I meant - I was meaning that giving a non-admin the rights to alter the whitelist was in effect giving them significant cluster-wide control and should be carefully considered. Rather than talking about granting cluster-admin rights which you've talked about here.  We definitely should talk about principle of least privilege and not handing our rights, but this should be in the security best practice we need to do for the ITHC outcomes.